WalletWasabi/WabiSabi/Client/AliceClient.cs

```diff
@@ -0,0 +1,101 @@
+using Microsoft.Extensions.Hosting;
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Bases;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+using WalletWasabi.WabiSabi.Backend.PostRequests;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	public class AliceClient : BackgroundService, IAsyncDisposable
+	{
+		private Guid AliceId { get; }
+		private Guid RoundId { get; }
+		private ArenaClient ArenaClient { get; }
+		public IEnumerable<ICoin> CoinsToRegister { get; }
```

Comment:
I would call it simply `Coins` because if we have an AliceClient instance that means the coins have already been registered.

Reply:
`ICoin` is an abstraction introduced in NBitcoin to represent _coins_  and _colored coin_ what means we will never need it. In fact, this code wouldn't work with colored coins because it casts `IMoney` to `Money` when colored coins use `MoneyBag` and no moey.

In summary, `Coin` is more correct here.

Reply:
ACK